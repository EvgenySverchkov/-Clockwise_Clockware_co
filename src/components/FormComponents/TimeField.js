import React, {useContext} from "react";
import PropTypes from "prop-types";

import Context from "../../ContextComponent";
import Tooltip from "../CustomTooltip";

const TimeField = ({ name, max, min, chngHandler, value }) => {
  const context = useContext(Context);
  return (
    <Tooltip 
    title={context.warningTooltipMsg} 
    open={context.isOpenWarningTooltip} 
    fieldId = {context.idForTooltip}
    tooltipIsOpen = {context.isOpenWarningTooltip}>
      <input
        type="time"
        name={name}
        max={max || null}
        min={min || null}
        onChange={chngHandler}
        value={value}
        id = "time"
        required
      />
    </Tooltip>
  );
}

TimeField.propTypes = {
  name: PropTypes.string.isRequired,
  max: PropTypes.string,
  min: PropTypes.string,
  chngHandler: PropTypes.func,
  value: PropTypes.string,
};

export default TimeField;
