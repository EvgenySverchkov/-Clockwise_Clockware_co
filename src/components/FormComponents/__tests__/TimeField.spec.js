import React from 'react';
import {create} from "react-test-renderer";

import TimeField from "../TimeField";
import ContextComponent from "../../../ContextComponent";

it("<TimeField /> test which is used in forms", ()=>{
    const mockProps = {
        name: "Name", 
        max: "18:00", 
        min: "09:00", 
        chngHandler: jest.fn(), 
        value: "Value"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <TimeField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
});