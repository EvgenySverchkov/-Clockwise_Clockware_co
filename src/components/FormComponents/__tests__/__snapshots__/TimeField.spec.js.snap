// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeField /> test which is used in forms 1`] = `
<input
  aria-describedby={null}
  className=""
  id="time"
  max="18:00"
  min="09:00"
  name="Name"
  onBlur={[Function]}
  onChange={[MockFunction]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  required={true}
  title="Message"
  type="time"
  value="Value"
/>
`;
